packages/shared_preferences/lib/shared_preferences_tizen.dart

```diff
@@ -26,51 +26,49 @@ class SharedPreferencesPlugin extends SharedPreferencesStorePlatform {
     final String key = pKey.toDartString();
     int ret;
 
-    final Pointer<Int8> pBool = malloc();
-    ret = bindings.getBoolean(pKey, pBool);
-    final bool boolValue = pBool.value == 1;
-    malloc.free(pBool);
-    if (ret == 0) {
-      _cachedPreferences![key] = boolValue;
-      return 1;
-    }
+    using((Arena arena) {
+      // Try as boolean type
+      final Pointer<Int8> pBool = arena();
+      ret = bindings.getBoolean(pKey, pBool);
+      final bool boolValue = pBool.value == 1;
+      if (ret == 0) {
+        _cachedPreferences![key] = boolValue;
+        return;
+      }
```

Comment:
If there's no need to call `malloc.free()`, the code can be simplified as

```suggestion
      if (bindings.getBoolean(pKey, pBool) == 0) {
        _cachedPreferences![key] = pBool.value == 1;
        return;
      }
```

Reply:
good point! :)